### PR TITLE
Weather my sql

### PR DESCRIPTION
**Pull request by OchirDarmaev**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -8,10 +10,9 @@
  import routerLogin from './routes/auth';
  import profile from './routes/profile';
  import routerSensors from './routes/sensors';
- import mongoose from 'mongoose';
+//  import mongoose from 'mongoose';
```
Comment (Cuprumbur) [inline src/index.ts]: not commit comment

File: src/routes/add.ts [hunk 2]
```diff
@@ -1,43 +1,35 @@
...
+router.get('/', (req, res) => {
     res.render('add', {
         title: "Новый датчик",
         isAdd: true
     });
 });
 
-router.post('/', sensorValidators, auth,   async (req:any, res:any) => {
-
+router.post('/', sensorValidators, auth, async (req:any, res:any) => {
```
Comment (Cuprumbur) [inline src/routes/add.ts]: consider use concrete type instead of `any`

File: src/routes/auth.ts [hunk 3]
```diff
@@ -13,31 +13,31 @@ const transporter = nodemailer.createTransport({
...
+        title: "Authorization",
         isLogin: true,
         RegisterError: req.flash('RegisterError'),
         LoginError: req.flash('LoginError')
     });
 });
 
 router.get('/auth/logout', async (req, res) => {
-    req!.session!.destroy(() => {
+     req.session!.destroy(() => {
```
Comment (Cuprumbur) [inline src/routes/auth.ts]: consider use typescript safe navigation operator `.?`

File: src/routes/auth.ts [hunk 4]
```diff
@@ -136,17 +134,18 @@ router.get('/auth/password/:tolen', async (req, res) => {
            return  res.redirect('/auth/reset');
         }
         try {
```
Comment (Cuprumbur) [inline src/routes/auth.ts]: why using  `try cath`

File: src/routes/auth.ts [hunk 5]
```diff
@@ -136,17 +134,18 @@ router.get('/auth/password/:tolen', async (req, res) => {
...
-        const candidate =  await User.findOne({
+        const candidate =  await User.findOne({ where: {
+
               resetTolen: req.params.tolen,
-              resetTolenExp: {$gt: Date.now()}
+              resetTolenExp: {$gt: Date.now()}}
         })
         if (!candidate) {
             res.redirect('/auth/login');
         } else {
```
Comment (Cuprumbur) [inline src/routes/auth.ts]: block `else` can be deleted
```
 if (!candidate) {
    res.redirect('/auth/login');
}
 
res.render('auth/password', {
    title: 'Restore access',
    error: req.flash('error'),
    userId: candidate._id.toString(),
    userId: candidate.id.toString(),
    tolen: req.params.tolen
})
```

File: src/routes/sensors.ts [hunk 6]
```diff
@@ -19,41 +22,63 @@ router.get('/sensors/edit/:id', auth, async (req, res) => {
...
-        detector
-    });
+    const detector = await Detector.findByPk(req.params.id);
+
+    if (detector !== null) {
+        const urlWeather = (`https://api.openweathermap.org/data/2.5/weather?q=${detector.location}&appid=${keys.API_URL}`);
+        const resp = await fetch(urlWeather);
+        try {
+            const mySity = await resp.json();
+            const temp = Math.trunc(mySity.main.temp - 273.15);
```
Comment (Cuprumbur) [inline src/routes/sensors.ts]: 273.15 move to const

File: src/routes/sensors.ts [hunk 7]
```diff
@@ -19,41 +22,63 @@ router.get('/sensors/edit/:id', auth, async (req, res) => {
...
-    await Detectors.findOneAndUpdate({_id: id}, req.body);
+    const location = req.body.location;
+    await Detector.update({location}, { where: {
+                           id  }
+                           });
     res.redirect('/sensors');
 })
 
 router.post('/delete', auth, async (req, res) => {
     try {
```
Comment (Cuprumbur) [inline src/routes/sensors.ts]: why using try cath

File: .vscode/settings.json [hunk 8]
```diff
@@ -1,6 +1,6 @@
 {
         // "terminal.integrated.shell.windows": "D:\\EPAM\\Git\\bin\\bash.exe"
-   "terminal.integrated.shell.windows": "C:\\Program Files\\Git\\bin\\bash.exe",
+   // "terminal.integrated.shell.windows": "C:\\Program Files\\Git\\bin\\bash.exe",
```
Comment (Cuprumbur) [inline .vscode/settings.json]: delete comment

File: public/index.css [hunk 9]
```diff
@@ -16,20 +16,82 @@ nav .brand-logo {
...
+.formCard {
+    display: inline;
+}
+
+a > p {
+    color:khaki;
+    font-size: 2em;
+}
+
+/*----------------------- ОШИБКИ ------------------------------ */
```
Comment (Cuprumbur) [inline public/index.css]: it can be deleted?

File: src/middleware/user.ts [hunk 10]
```diff
@@ -5,6 +5,6 @@ export default async (req: Request, res: Response, next: NextFunction) => {
     if (!req!.session!.user) {
         return next();
     }
-    req.body.user =  await User.findById(req!.session!.user._id);
+    req.body.user =  await User.findByPk(req!.session!.user.id);
```
Comment (Cuprumbur) [inline src/middleware/user.ts]: consider use typescript safe navigation operator

File: .vscode/settings.json (hunk 8 above)
Comment (IlyaIvanchikov) [reply .vscode/settings.json]: delete //

**Pull request closed without merging**